src/FixedDataTableCellGroup.js

```diff
@@ -74,59 +79,20 @@ class FixedDataTableCellGroupImpl extends React.Component {
 
   render() /*object*/ {
     const props = this.props;
-    const columnsToRender = props.columnsToRender;
     const columns = props.columns;
-    const contentWidth = sumPropWidths(columns);
+    const columnGroupWidth = sumPropWidths(columns);
 
     const isColumnReordering = props.isColumnReordering && columns.reduce(function (acc, column) {
       return acc || props.columnReorderingData.columnKey === column.props.columnKey;
     }, false);
 
-    if (this.props.isScrolling) {
-      // We are scrolling, so there's no need to display any cells which lie outside the viewport.
-      // We still need to render them though, so as to not cause any unmounts.
-      this._staticCellArray.forEach((cell, i) => {
-        this._staticCellArray[i] = React.cloneElement(this._staticCellArray[i], { visible: false });
-      });
-    } else {
-      // reset the static cell array if scrolling is stopped
-      // this is done so that only cells inside the buffer are considered for vertical scrolling
-      this._staticCellArray = [];
-    }
-
-    for (let i = 0; i < props.columns.length; i++) {
-      const columnIndex = columnsToRender === undefined ? i : columnsToRender[i];
-
-      // if columnIndex doesn't exist, it means that the cell isn't visible or that it doesn't exist at all
-      if (columnIndex === undefined) {
-        continue;
-      }
-
-      const columnProps = columns[columnIndex].props;
-      const cellTemplate = columns[columnIndex].template;
-      const key = columnProps.columnKey || 'cell_' + i;
-
-      const currentPosition = props.columnOffsets[columnIndex];
-      const visible = currentPosition - props.left <= props.width &&
-        currentPosition - props.left + columnProps.width >= 0;
-
-      this._staticCellArray[i] = this._renderCell(
-        props.rowIndex,
-        props.rowHeight,
-        columnProps,
-        cellTemplate,
-        currentPosition,
-        visible,
-        key,
-        contentWidth,
-        isColumnReordering
-      );
-    }
+    this._computeVirtualizedCells({ isColumnReordering }); // mutates this._staticCellArray
```

Comment:
refactored here

* computeVirtualizedCells will get list of cells inside the viewport that can be _virtualized_ (aka `allowCellsRecycling` is turned on) into this._staticCellArray
* nonVirtualizedCells will get list of all cells inside the viewport that cannot be virtualized (`allowCellsRecycling` is turned off) and returns it